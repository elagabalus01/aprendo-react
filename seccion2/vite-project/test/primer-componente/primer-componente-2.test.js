import {getByAltText, render, screen} from "@testing-library/react";
import PrimerComponente from "../../src/primer-componente/primer-componente.jsx";

describe("prueba primer componetne", ()=>{

    const title = "ESTE ES EL TITULO";
    const subtitulo = "EL SUBTITULO"

    test("Debe hacer match con el snapshot", ()=>{
        const {container} = render(<PrimerComponente title={title}></PrimerComponente>)

        expect(container).toMatchSnapshot()
    })

    test("Debe mostrar el mensaje hola mudno", ()=>{
        render(<PrimerComponente title={title}></PrimerComponente>)
        expect(screen.getByText(title)).toBeTruthy()
    })


    test("Debe mostrar el titulo en un h1", ()=>{
        render(<PrimerComponente title={title}></PrimerComponente>)

        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })


    test("Debe mostrar el subtitulo en un h2", ()=>{
        render(<PrimerComponente title={title} subtitle={subtitulo}></PrimerComponente>)

        expect(screen.getAllByText(subtitulo).length).toBe(1)
    })


})
