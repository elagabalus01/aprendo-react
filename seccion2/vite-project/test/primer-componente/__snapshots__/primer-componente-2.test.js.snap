// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba primer componetne Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola mundo cruel desde una funci√≤n
  </h2>
  <h1>
    ESTE ES EL TITULO
  </h1>
  <p>
    Hola mundo
  </p>
</div>
`;
